src/com/android/settings/wifi/WifiTimeoutSettings.java

```diff
@@ -0,0 +1,164 @@
+package com.android.settings.wifi;
+
+import android.app.settings.SettingsEnums;
+import android.content.Context;
+import android.content.pm.PackageManager;
+import android.graphics.drawable.Drawable;
+import android.provider.Settings;
+import android.text.format.DateUtils;
+
+import androidx.preference.PreferenceScreen;
+
+import com.android.settings.R;
+import com.android.settings.search.BaseSearchIndexProvider;
+import com.android.settings.widget.RadioButtonPickerFragment;
+import com.android.settingslib.search.SearchIndexable;
+import com.android.settingslib.widget.CandidateInfo;
+import com.android.settingslib.widget.RadioButtonPreference;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@SearchIndexable(forTarget = SearchIndexable.ALL & ~SearchIndexable.ARC)
+public class WifiTimeoutSettings extends RadioButtonPickerFragment {
+    public static final BaseSearchIndexProvider SEARCH_INDEX_DATA_PROVIDER =
+            new BaseSearchIndexProvider(R.xml.wifi_configure_settings) {
+                @Override
+                protected boolean isPageSearchEnabled(Context context) {
+                    return context.getPackageManager().hasSystemFeature(
+                            PackageManager.FEATURE_WIFI);
+                }
+            };
+    Context mContext;
+    enum TimeoutEnum{
+
+        DISABLED(0),
+        SEC_FIFTEEN(15000),
+        SEC_THIRTY(30000),
```

Comment:
`java.util.concurrent.TimeUnit` is usually used in such cases:

`TimeUnit.MINUTES.toMillis(5)`,
`TimeUnit.HOURS.toMillis(2)`
etc

I think a better code style would be to declare a `long` array of possible timeouts:
```
static final long[] CANDIDATES = {
    0,
    TimeUnit.SECONDS.toMillis(15),
    ...
    TimeUnit.HOURS.toMillis(8),
};
```

Reply:
done.